src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java

```diff
@@ -177,9 +178,10 @@ public void openLogTab() {
   }
 
   public void setFocusOnNewCode(boolean isFocusOnNewCode) {
-    this.<CurrentFilePanel>updateTab(SonarLintToolWindowFactory.CURRENT_FILE_TAB_TITLE, panel -> panel.setFocusOnNewCode(isFocusOnNewCode));
-    // TODO
-    // this.updateTab(SonarLintToolWindowFactory.REPORT_TAB_TITLE, ReportPanel::setFocusOnNewCode);
+    this.<CurrentFilePanel>updateTab(SonarLintToolWindowFactory.CURRENT_FILE_TAB_TITLE, panel -> panel.setFocusOnNewCode(getService(project, CleanAsYouCodeService.class)
+      .shouldFocusOnNewCode()));
+    this.<ReportPanel>updateTab(SonarLintToolWindowFactory.REPORT_TAB_TITLE, panel -> panel.setFocusOnNewCode(getService(project, CleanAsYouCodeService.class)
+      .shouldFocusOnNewCode()));
```

Comment:
I was about to comment that you could introduce a variable to not call the service twice. In fact I think we need to rework the whole method, the parameter is probably useless